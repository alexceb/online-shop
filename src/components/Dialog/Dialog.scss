@import '../../resources/vars.scss';

.dialogContainer {
  display: flex;
  position: relative;
  align-items: center;
  justify-content: center;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 2;
  background-color: #ccc;
  opacity: 0.5;

  .dialogContent {
    display: inline-block;
    width: 15rem;
    height: 10rem;
    background-color: #fff;
    text-align: center;

    & > button {
      display: inline-block;
      min-width: 9rem;
      padding: 0.4rem 0;
      margin: 0 0.5rem;
      font-size: 1rem;
      border-radius: 0.3rem;
      cursor: pointer;
      border: none;
      color: #fff;
      box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
    }

    .cartButton {
      background-color: $shop-default-green;
    }
    
    .continueShopping {
      background-color: $shop-gray;
    }
  }
}