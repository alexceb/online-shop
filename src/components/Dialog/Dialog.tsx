import * as React from 'react';
import * as styles from './Dialog.scss';

export interface DialogProps {
  visible: boolean
  onGoShopping: () => void
  onGoCart: () => void
}

export const Dialog: React.FC<DialogProps> = props => {
  const { visible, onGoShopping, onGoCart } = props;

  return (
    <>
      {visible ? (
        <div className={styles.dialogContainer}>
          <div className={styles.dialogContent}>
            {/* <a href="#" className={styles.closeButton} aria-hidden="true"
               onClick={(e) => onCloseDialog(e)}>&times;</a> */}
            <div className={styles.dialogMessage}>You successfully added product to cart</div>
            <div className={styles.bottom}>
              <button className={styles.continueShopping}
                      onClick={() => onGoShopping()}>
                Continue shopping
              </button>
              <button className={styles.cartButton}
                      onClick={() => onGoCart()}>
                Go to cart
              </button>
            </div>
          </div>
        </div>
      ) : null}
    </>
    
  )
};