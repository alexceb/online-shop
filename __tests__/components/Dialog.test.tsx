import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Dialog, DialogProps } from '../../src/components/Dialog/Dialog';

describe('Dialog notification component', () => {
  describe('render()', () => {
    test('Renders <Dialog /> correctly', () => {
      const props: DialogProps = {
        visible: true,
        onGoShopping: jest.fn(),
        onGoCart: jest.fn(),
      };
      const wrapper = shallow(<Dialog {...props} />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('It renders message to the user', () => {
      const props: DialogProps = {
        visible: true,
        onGoShopping: jest.fn(),
        onGoCart: jest.fn(),
      };
      const wrapper = shallow(<Dialog {...props} />);
      const expectedMessage = 'You successfully added product to cart';

      expect(wrapper.find('.dialogMessage').text()).toBe(expectedMessage);
    });
  });

  describe('It should call a function on button click', () => {
    test('It should invoke onGoShopping function on button click', () => {
      const props: DialogProps = {
        visible: true,
        onGoShopping: jest.fn(),
        onGoCart: jest.fn(),
      };
      const wrapper = shallow(<Dialog {...props} />);

      wrapper.find('button.continueShopping').simulate('click');

      expect(props.onGoShopping).toHaveBeenCalled();
    });

    test('It should invoke onGoToCart function on button click', () => {
      const props: DialogProps = {
        visible: true,
        onGoShopping: jest.fn(),
        onGoCart: jest.fn(),
      };
      const wrapper = shallow(<Dialog {...props} />);

      wrapper.find('button.cartButton').simulate('click');

      expect(props.onGoCart).toHaveBeenCalled();
    })
  });

  describe('Render with visible set to false', () => {
    test('It should not render dialog content if visible set to false', () => {
      const props: DialogProps = {
        visible: false,
        onGoShopping: jest.fn(),
        onGoCart: jest.fn(),
      };
      const wrapper = shallow(<Dialog {...props} />);

      expect(wrapper.find('.dialogContainer')).toHaveLength(0);
    });
  });
});