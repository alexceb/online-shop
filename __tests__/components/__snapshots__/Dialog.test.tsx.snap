// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog notification component render() Renders <Dialog /> correctly 1`] = `
<Fragment>
  <div
    className="dialogContainer"
  >
    <div
      className="dialogContent"
    >
      <div
        className="dialogMessage"
      >
        You successfully added product to cart
      </div>
      <div
        className="bottom"
      >
        <button
          className="continueShopping"
          onClick={[Function]}
        >
          Continue shopping
        </button>
        <button
          className="cartButton"
          onClick={[Function]}
        >
          Go to cart
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;
